integration/README.md

```diff
@@ -1,3 +1,66 @@
 # Integration
 
-A library for writing integration tests against providers.
+
+The `integration` package provides a library for writing integration tests for your provider code. It enables testing provider behavior in-memory, sitting just above the gRPC level. This package is particularly useful for validating the lifecycle of resources and ensuring correctness in provider implementations.
+
+## Server
+
+The package includes a `Server` interface and its implementation, which acts as a test harness for Pulumi providers.
+The interface allows you to exercise the RPC methods of your provider, including:
+- Schema retrieval
+- Provider configuration
+- Resource lifecycle operations (create, read, update, delete)
+- Component lifecycle operations (construct)
+- Provider functions (invoke) and resource functions (call)
+
+To make a server, call `integration.NewServerWithOptions` and pass the `Provider` instance to be tested.
+Then, exercise the various RPC methods of your provider.
+
+## Component Resource Mocks
+
+Since the business logic of a component typically creates child resources, testing the logic usually involves
+mocking the implementation of child resources.  For example, if your component creates a `random.RandomPet` resource,
+you can use a mock to return simulated resource state, i.e. the `RandomPet.ID` property. The actual random provider
+is never called.
+
+To configure mocking, use the `integration.WithMocks` server option and pass an implementation of `pulumi.MockResourceMonitor`.
+The mock monitor receives a callback for the component resource and for each child resource as it is registered,
+giving you an opportunity to return a simulated state for each child. See `integation.MockMonitor` for a simple implementation.
+
+To test a component resource, call the `Construct` method on the integration server.
```

Comment:
It would be great to have an example of testing component resources here. If you don't have time to add an example, can you create a follow up issue for that?